### PR TITLE
✨ snowflake: resource monitors, auth/masking policies, policy references, secrets

### PR DESCRIPTION
**Pull request by tas50**
## Summary

Expands the snowflake provider with five resources that target the highest-value security audits not previously expressible:

- **`snowflake.resourceMonitor`** — credit-usage guardrails (quota, used/remaining credits, level, notify/suspend thresholds, notify-user list).
- **`snowflake.authenticationPolicy`** — authentication-method, MFA, and client-type controls. Definitive answer to "is MFA enforced" and "is password auth disabled". `DESCRIBE` properties (`authenticationMethods`, `mfaAuthenticationMethods`, `mfaEnrollment`, `clientTypes`, `securityIntegrations`) lazy-fetched.
- **`snowflake.maskingPolicy`** — column-level masking policies; `signature`, `returnType`, and `body` lazy-fetched via `DESCRIBE`.
- **`snowflake.policyReference`** (private) — a single policy attachment row, reachable via `maskingPolicy.references`. Backs the "is this masking policy actually attached to a column?" audit by querying `INFORMATION_SCHEMA.POLICY_REFERENCES`.
- **`snowflake.secret`** — Snowflake-managed secrets (PASSWORD, OAUTH2, GENERIC_STRING) with lazy `DESCRIBE` for username, integration name, and OAuth token expiry timestamps.

Account-level enumeration: `snowflake.account.resourceMonitors`, `.authenticationPolicies`, `.maskingPolicies`, `.secrets`.

### Notable design choices
- `policyReference` is `private` (no top-level catalog entry) and reached via `maskingPolicy.references`. Future `.references` accessors on row-access / authentication policies can reuse it.
- `authenticationPolicy.createdAt` is `string` because the SDK exposes it that way; matches existing patterns like `failoverGroup.nextScheduledRefresh`.
- The `INFORMATION_SCHEMA.POLICY_REFERENCES` raw SQL escapes the database name (`"` → `""`) and policy name (`'` → `''`).
- Unit-tested `parseAuthPolicyList` for the paren-quoted `DESCRIBE` output parsing edge cases (empty, single, multiple, quoted, whitespace).

## Test plan

- [ ] `mql shell snowflake -c "snowflake.account.resourceMonitors { name level creditQuota usedCredits notifyAt suspendAt }"`
- [ ] `mql shell snowflake -c "snowflake.account.authenticationPolicies { name mfaEnrollment authenticationMethods clientTypes securityIntegrations }"`
- [ ] `mql shell snowflake -c "snowflake.account.maskingPolicies { name signature returnType body references { refEntityDomain refEntityName refColumnName policyStatus } }"`
- [ ] `mql shell snowflake -c "snowflake.account.secrets { name secretType integrationName username oauthAccessTokenExpiryTime }"`
- [ ] Verify that a masking policy with no current attachments returns an empty `references` list (not an error).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: providers/snowflake/resources/secrets.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	}
+	return r.descAccessExp, nil
+}
+
+func (r *mqlSnowflakeSecret) oauthRefreshTokenExpiryTime() (*time.Time, error) {
+	if err := r.gatherDescribe(); err != nil {
+		return nil, err
+	}
+	return r.descRefreshExp, nil
+}
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/snowflake/resources/secrets.go:151]: Leaving as-is: CLAUDE.md's StateIsNull rule is explicitly scoped to singular *resource* pointer returns — `*time.Time` is a basic-type pointer that falls outside the rule, and existing time accessors in the codebase (e.g. `providers/aws/resources/aws_keyspaces.go` `earliestRestorableTimestamp`, `lastUpdateToPayPerRequestTimestamp`) return `nil, nil` without setting StateIsNull. If we want to move to the defensive pattern, we should do it for all time accessors in a separate sweep rather than diverging here.

File: providers/snowflake/resources/secrets.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		return "", err
+	}
+	return r.descIntegration, nil
+}
+
+func (r *mqlSnowflakeSecret) oauthAccessTokenExpiryTime() (*time.Time, error) {
+	if err := r.gatherDescribe(); err != nil {
+		return nil, err
+	}
+	return r.descAccessExp, nil
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/snowflake/resources/secrets.go:139]: Applied — set `StateIsSet | StateIsNull` before returning `nil, nil` on both expiry accessors in 3413ff968.
